### DISCUSSION TITLE
Inconsistent Table Detection

### DISCUSSION THREAD

**Discussion by dlasusa** [jsvine/pdfplumber#1131] [Ask for help with specific PDFs]
I don't think I can share the pdf since it's got police data in it, but I'm sharing images of the issue (I've removed some info from the images, but I don't think it's relevant).  

In the PDF, when a page has multiple entries, it seems to detect the tables as expected.  But if there is a single item, it puts a box around the item, but I don't get the blue "table" color.

What I expect:
![pdf_sample2](https://github.com/jsvine/pdfplumber/assets/36902797/f4a19a71-bdb3-490b-a43a-b8da4ee9034a)

When there is just one item:
![pdf_sample](https://github.com/jsvine/pdfplumber/assets/36902797/4fb8a960-0e50-4e29-b13f-bb098d962305)

```py
import pdfplumber

table_settings = {
    "explicit_vertical_lines": [17, 528],
    "horizontal_strategy": "lines",
    "snap_y_tolerance": 10,
    "intersection_tolerance": 10,
    "join_tolerance": 300
}

with pdfplumber.open('file.pdf') as pdf:
    first_page = pdf.pages[1].to_image(resolution=150)
    first_page.reset().debug_tablefinder(table_settings).show()
```
Environment:
Win11
Python 3.12.2
pdfplumber  0.11.0

**Comment by jsvine** [comment]
Hi @dlasusa, and thanks for sharing this example. The bad news: `pdfplumber` considers something a table only if it contains at least two cells, in large part to avoid any random rectangle from appearing as a "table". The good news: The "cells" in the "tables" in this PDF are clearly demarcated by the only horizontal lines on the pages. So I'd suggest a strategy along the lines of:

- Identify all the horizontal lines on the page via `page.horizontal_edges`
- Iterate between each pair of lines and use `page.crop(...).extract_text()` to grab the text between them

**Comment by dlasusa** [comment]
Thanks @jsvine !  I ended up iterating the pdf twice (with 2 different sets of table settings).  The pass through I tracked which pages had no results, and the second pass, iterated just those pages and combined the results.  I wasn't sure if I was doing something wrong.  My first pass table settings:
```py
table_settings = {
    "explicit_vertical_lines": [17, 528],
    "horizontal_strategy": "lines",
    "snap_y_tolerance": 10,
    "intersection_tolerance": 10,
    "join_tolerance": 300
}
```
Second pass:
```py
table_settings2 = {
    "explicit_vertical_lines": [17, 528],
    "explicit_horizontal_lines": [60, 190],
    "horizontal_strategy": "lines",
    "snap_tolerance": 10,
    "intersection_tolerance": 10,
    "join_tolerance": 300
}
```
By adding some `explicit_horizontal_lines` I think I stumbled into creating extra "blank tables" so there was more than one table, at which point I simply had to ignore the blank tables and pull the real data out (hopefully that made sense ha!)  Probably not the prettiest solution, but it got the job done. 

Thank you for your suggestion and your time!

**Reply by jsvine** [reply]
Thanks for the update!
